apps/web/src/quoter/utils/edgePoolQueries.ts

```diff
@@ -0,0 +1,317 @@
+  const currencyA = mockCurrency(addressA, chainId)
+  const currencyB = mockCurrency(addressB, chainId)
+
+  const pools = await InfinityRouter.getV3CandidatePools({
+    currencyA,
+    currencyB,
+    clientProvider: getProvider(),
+  })
+
+  const chain = getChainName(chainId)
+  const tvlMap = await poolTvlMap(['v3'], chain as APIChain)
+  const poolsWithTvl = fillTvl(tvlMap, pools) as V3PoolWithTvl[]
+  return SmartRouter.v3PoolTvlSelector(currencyA, currencyB, poolsWithTvl)
+}
+
+const fetchSSPool = async (addressA: Address, addressB: Address, chainId: ChainId) => {
+  const currencyA = mockCurrency(addressA, chainId)
+  const currencyB = mockCurrency(addressB, chainId)
+  const client = getProvider()
+  const blockNumber = await client({ chainId })?.getBlockNumber()
+
+  const pools = await SmartRouter.getStableCandidatePools({
+    currencyA,
+    currencyB,
+    onChainProvider: getProvider(),
+    blockNumber,
+  })
+
+  const chain = getChainName(chainId)
+  const tvlMap = await poolTvlMap(['stable'], chain as APIChain)
+  return fillTvl(tvlMap, pools) as StablePoolWithTvl[]
+}
+
+const querySingleType = async (chainId: ChainId, protocol: Protocol, addressA: Address, addressB: Address) => {
+  switch (protocol) {
+    case 'v2': {
+      return fetchV2Pools(addressA, addressB, chainId)
+    }
+    case 'stable': {
+      return fetchSSPool(addressA, addressB, chainId)
+    }
+    case 'v3': {
+      return fetchV3Pools(addressA, addressB, chainId)
+    }
+    case 'infinityBin':
+    case 'infinityCl': {
+      return fetchInfinityPools(addressA, addressB, chainId)
+    }
+    default:
+      throw new Error('invalid pool')
+  }
+}
+const fetchAllCandidatePools = async (
+  addressA: Address,
+  addressB: Address,
+  chainId: ChainId,
+  protocols: Protocol[],
+) => {
+  const queries = await Promise.all(
+    protocols
+      .filter((x) => x !== 'infinityBin') // For infinity pools fetch together
+      .map((protocol) => querySingleType(chainId, protocol as Protocol, addressA, addressB)),
+  )
+  const pools = queries.flat() as (InfinityPoolWithTvl | V2PoolWithTvl | V3PoolWithTvl | StablePoolWithTvl)[]
+  return pools.map((pool) => {
+    return SmartRouter.Transformer.serializePool(pool as Pool)
+  })
+}
+
+function fillTvl(tvlMap: Record<`0x${string}`, string>, pools: Pool[]) {
+  return pools.map((pool) => {
+    const id = getPoolAddress(pool)
+    const tvlUSD: string = tvlMap[id] || '0'
+    const bigIntTvlUSD = BigInt(Math.floor(Number(tvlUSD)))
+    if ('tvlUSD' in pool) {
+      return { ...pool, tvlUSD: bigIntTvlUSD }
+    }
+    return pool as Pool & WithTvl
+  })
+}
+
+export const poolTvlMap = async (protocols: Protocol[], chain: APIChain) => {
+  try {
+    const remotePools = await fetchAllPools({
+      baseUrl: 'https://explorer.pancakeswap.com/api/cached/pools/tvl-refs',
+      protocols,
+      chains: [chain],
+      orderBy: 'tvlUSD',
+      pageSize: 100,
```

Comment:
page size can be greater like 1000 if the purpose of it is to fetch all the existing pools